### PR TITLE
Refactor if ale customizing to abstract class

### LINKED ISSUE #30: Convert customizing interface to abstract class

### PR DESCRIPTION
**Pull request by Goala**
https://github.com/Goala/abap-log-exporter/issues/30

### CODE REVIEW

File: src/zcl_ale_customizing.clas.abap [hunk 1]
```diff
@@ -1,59 +1,28 @@
...
-                 connector_class TYPE zale_config-ale_key VALUE 'CNCTR_CLAS',
-               END OF customzing_fields.
-
-  PROTECTED SECTION.
-  PRIVATE SECTION.
-
-ENDCLASS.
-
+    METHODS set_scenario ABSTRACT
+      IMPORTING scenario TYPE zale_config-ale_scenario.
```
Comment (JohannesKonings) [inline src/zcl_ale_customizing.clas.abap]: Could you change that to a constructor? Than it'll not forgotten to set the scenario.

File: src/zcl_ale_customizing_base.clas.abap [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  METHOD set_scenario.
+
+    me->scenario = scenario.
+
+    SELECT *
+    FROM zale_config
+    INTO TABLE @configurations
+    WHERE ale_scenario = @scenario.
+
+  ENDMETHOD.
```
Comment (JohannesKonings) [inline src/zcl_ale_customizing_base.clas.abap]: That should be in the abstract class, than it could be used for all inheriting classes.

**Review by JohannesKonings — CHANGES_REQUESTED** [review]
see comments

File: src/zcl_ale_customizing.clas.abap (hunk 1 above)
Comment (Goala) [reply src/zcl_ale_customizing.clas.abap]: implemented

File: src/zcl_ale_customizing_base.clas.abap (hunk 2 above)
Comment (Goala) [reply src/zcl_ale_customizing_base.clas.abap]: implemented

**Review by JohannesKonings — APPROVED** [review]
LGTM

**Pull request merged**